packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -102,9 +104,15 @@ export abstract class AbstractResourcePreferenceProvider extends PreferenceProvi
             return true;
         }
         try {
+            const editor = await this.editorManager.getByUri(new URI(resourceUri!));
```

Comment:
What is there is no open editor? Should not we use preferences then?

Reply:
Yes, I think that'd be a better fallback.
I thought of it initially, if no open editors are present, use the preference value for JSON editors.

Reply:
So we agree on it?

I generally don't like that we couple it here with the editor extension, maybe we always rely on preferences? by using preference service and strings as identifires? If an editor extension is not installed then we can fall back to `2` as tabSize  and `true` as insertSpaces.

Reply:
> So we agree on it?
> 
> I generally don't like that we couple it here with the editor extension, maybe we always rely on preferences? by using preference service and strings as identifires? If an editor extension is not installed then we can fall back to `2` as tabSize and `true` as insertSpaces.

Yes, for simplicity I'll update the PR and only use the preferences instead.
I won't check for the editor and attempt to determine the current indentation and type.